File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.cs
public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        if(nums == null) return 0.00;
        
        int left = 0, right = 0, currSum = 0, maxSum = Int32.MinValue;
        while(right < nums.Length){
            currSum += nums[right];
            right++;
                
            if(right - k > 0){
                currSum -= nums[left];
                left++;
            }
            
            if(right - left == k){
                maxSum = Math.Max(maxSum, currSum);
            }
                
        }
        
        return (double)maxSum / k;
    }
}

File: 0643-maximum-average-subarray-i/0643-maximum-average-subarray-i.ts
function findMaxAverage(nums: number[], k: number): number {
    let sum: number=0;
    
    for(let i=0; i < k; i++){
        sum += nums[i];        
    }

    let ret: number = sum;
    for(let i=k; i < nums.length;i++){
        sum += (nums[i]-nums[i-k]);
        ret = Math.max(sum, ret);        
    }
    
    return ret / k;
    
};